src/celeritas/ext/GeantSetup.cc

```diff
@@ -67,6 +68,22 @@ class DetectorConstruction : public G4VUserDetectorConstruction
 //---------------------------------------------------------------------------//
 }  // namespace
 
+//---------------------------------------------------------------------------//
+/*!
+ * G4RunManager::GetNumberOfThreads isn't virtual before Geant4 v10.7.0 so we
+ * need to explicitely try dynamic cast to G4MTRunManager to get the number of
+ * threads. If the dynamic cast fails, return 1
+ */
+int GetNumberOfThreads(G4RunManager const& runman)
+{
+    auto const* runman_mt = dynamic_cast<G4MTRunManager const*>(&runman);
+    if (runman_mt != nullptr)
```

Comment:
Minimizing the scope of temporary variables is good; the following is idiomatic C/C++:
```suggestion
    if (auto const* runman_mt = dynamic_cast<G4MTRunManager const*>(&runman))
```